litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/index.tsx

```diff
@@ -175,6 +132,60 @@ const TuneWorkflow = forwardRef((_, ref) => {
   const [generatedYAML, setGeneratedYAML] = useState<CustomYAML>(
     manifest === '' ? yamlTemplate : YAML.parse(manifest)
   );
+
+  // This function fetches the manifest for pre-defined workflows
+  const fetchYaml = (link: string) => {
+    fetch(link)
+      .then((data) => {
+        data.text().then((yamlText) => {
+          setGeneratedYAML(YAML.parse(yamlText));
+          workflowAction.setWorkflowManifest({
+            manifest: yamlText,
+          });
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  // Index DB Fetching for extracting selected Button and Workflow Details
+  const getSelectedWorkflowDetails = () => {
+    localforage.getItem('workflow').then((workflow) =>
+      setWorkflow({
+        name: (workflow as WorkflowDetailsProps).name,
+        crd: (workflow as WorkflowDetailsProps).CRDLink,
+        description: (workflow as WorkflowDetailsProps).description,
+      })
+    );
+    localforage.getItem('selectedScheduleOption').then((value) => {
+      // Setting default data when MyHub is selected
+      if (value !== null && (value as ChooseWorkflowRadio).selected === 'A') {
+        setCustomWorkflow(false);
+        localforage.getItem('workflow').then((value) => {
+          if (value !== null && (value as WorkflowDetailsProps).CRDLink !== '')
+            fetchYaml((value as WorkflowDetailsProps).CRDLink);
+        });
+      }
+      if (value !== null && (value as ChooseWorkflowRadio).selected === 'C') {
+        setCustomWorkflow(true);
+        localforage.getItem('selectedHub').then((hub) => {
+          setHubName(hub as string);
+          getCharts({
+            variables: { projectID: selectedProjectID, HubName: hub as string },
+          });
+        });
+      }
+      if (value !== null && (value as ChooseWorkflowRadio).selected === 'D') {
```

Comment:
use constants for A, C, D

Reply:
I believe we dont need separate constants for just A,B,C,D. There are just to check the data present in the localforage. cc: @S-ayanide